codes/c/chapter_dynamic_programming/min_cost_climbing_stairs_dp.c

```diff
@@ -0,0 +1,65 @@
+/**
+ * File: min_cost_climbing_stairs_dp.c
+ * Created Time: 2023-10-02
+ * Author: Zuoxun (845242523@qq.com)
+ */
+
+#include "../utils/common.h"
+
+int cost_size;
+
+/* 爬楼梯最小代价：动态规划 */
+int minCostClimbingStairsDP(int cost[]) {
+    int n = cost_size - 1;
+    if (n == 1 || n == 2)
+        return cost[n];
+    // 初始化 dp 表，用于存储子问题的解
+    int dp[n + 1];
+    // 初始状态：预设最小子问题的解
+    dp[1] = cost[1];
+    dp[2] = cost[2];
+    // 状态转移：从较小子问题逐步求解较大子问题
+    for (int i = 3; i <= n; i++) {
+        dp[i] = min(dp[i - 1], dp[i - 2]) + cost[i];
```

Comment:
There is no `min` declaration and implementation in the public header file of the c code. If you use it directly here, you should get an error.

Reply:
I've declared function min in this file.

Reply:
> I've declared function min in this file.

I don’t see the `min` forward declaration before using the min function.

Reply:
Sorry, i didn't declare it before using it, i'll edit tomorrow.